classes/activities.php

```diff
@@ -0,0 +1,87 @@
+<?php
+// This file is part of Moodle - https://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <https://www.gnu.org/licenses/>.
+
+namespace mod_learningmap;
+
+use completion_info;
+use cm_info;
+use stdClass;
+
+/**
+ * Class for handling the activities of a learning map
+ *
+ * @package     mod_learningmap
+ * @copyright   2021-2023, ISB Bayern
+ * @author      Stefan Hanauska <stefan.hanauska@csg-in.de>
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class activities {
```

Comment:
I'm not sure about the purpose of this class. Is this more a utility class? Or should this represent a single activity of the learningmap.

If it's the first:
1. Either rename it to `activitymanager`
2. Or rename it to `activityutils`, in this case I would suggest to make the `is_completed` method static and pass all the variables needed directly to it

If it's the second: Suggestion: Renaming to `activity` or `learningmapactivity`. In this case I would also pass the cm_info to the constructor or - if possible (but I do not think so, because we need the cm_info object from the `\core_completion\activity_custom_completion` object we inherit from) - create it in the constructor.
The class then would represent a learningmap activity, maybe can be extended later on to store and manage many more attributes of an activity. The completion state would only be one of them.

Reply:
This class helps accessing the activities contained in the learning map. The cm_info object passed to is_completed is not the cm_info object referencing the learning map itself - it references the activity we are checking for completion.
Right now, it only handles completion, but there will be more methods in future versions. Using a static method would mean I'd have to get completion_info outside of the class to avoid retrieving it for every single call (for bigger maps, this could impact performance).

Reply:
Renamed to activitymanager